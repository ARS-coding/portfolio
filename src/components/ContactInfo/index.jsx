import React from 'react';
import { Row, Col, OverlayTrigger, Popover, Button } from 'react-bootstrap';
import './index.css';

const ContactInfo = () => {
  const phonePopover = (
    <Popover id='popover-basic'>
      <Popover.Content>
        <Button href='tel:+905523600893' variant='link'>
          +90 552 360 08 93
        </Button>
      </Popover.Content>
    </Popover>
  );

  const emailPopover = (
    <Popover id='popover-basic'>
      <Popover.Content>
        <Button href='mailto: mansuroglu99@gmail.com' variant='link'>
          mansuroglu99@gmail.com
        </Button>
      </Popover.Content>
    </Popover>
  );

  return (
    <Row className='contactInfo'>
      <Col xs={4}>
        <a
          target='_blank'
          rel='noreferrer'
          href='https://www.linkedin.com/in/tarkanmansuroglu/'
        >
          <i className='fab fa-linkedin-in fa-3x contactLinkedIn'></i>
        </a>
      </Col>
      <Col xs={4}>
        <a
          target='_blank'
          rel='noreferrer'
          href='https://github.com/tmansuroglu'
        >
          <i className='fab fa-github fa-3x contactGithub'></i>
        </a>
      </Col>
      <Col xs={4}>
        <a
          target='_blank'
          rel='noreferrer'
          href='https://www.instagram.com/tarkanmansuroglu/'
        >
          <i className='fab fa-instagram fa-3x contactInstagram'></i>
        </a>
      </Col>
      <Col xs={4}>
        <a
          target='_blank'
          rel='noreferrer'
          href='https://twitter.com/T_Mansuroglu'
        >
          <i className='fab fa-twitter fa-3x contactTwitter'></i>
        </a>
      </Col>
      <Col xs={4}>
        <OverlayTrigger
          trigger='click'
          placement='bottom'
          overlay={phonePopover}
          rootClose={true}
          rootCloseEvent='click'
        >
          <i className='fas fa-mobile-alt fa-3x'></i>
        </OverlayTrigger>
      </Col>
      <Col xs={4}>
        <OverlayTrigger
          trigger='click'
          placement='bottom'
          overlay={emailPopover}
          rootClose={true}
          rootCloseEvent='click'
        >
          <i className='far fa-envelope fa-3x'></i>
        </OverlayTrigger>
      </Col>
    </Row>
  );
};

export default ContactInfo;
