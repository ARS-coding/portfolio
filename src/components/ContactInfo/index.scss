.contactLinkedIn {
  color: #2867b2;
}

.contactInfo .col-4 {
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  padding: 12px;
}

.contactTwitter {
  color: #1da1f2;
  margin: auto;
}

.contactGithub {
  color: #333;
}
.contactInstagram {
  background: #d6249f;
  background: radial-gradient(
    circle at 30% 107%,
    #fdf497 0%,
    #fdf497 5%,
    #fd5949 45%,
    #d6249f 60%,
    #285aeb 90%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}
