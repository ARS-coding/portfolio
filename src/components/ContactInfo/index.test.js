import React from 'react';
import renderer from 'react-test-renderer';
import ContactInfo from '.';

it('ContactInfo renders correctly', () => {
  const tree = renderer.create(<ContactInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
