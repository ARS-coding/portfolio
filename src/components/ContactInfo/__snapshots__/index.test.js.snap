// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo renders correctly 1`] = `
<div
  className="contactInfo row"
>
  <div
    className="col-4"
  >
    <a
      href="https://www.linkedin.com/in/tarkanmansuroglu/"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-linkedin-in fa-3x contactLinkedIn"
      />
    </a>
  </div>
  <div
    className="col-4"
  >
    <a
      href="https://github.com/tmansuroglu"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github fa-3x contactGithub"
      />
    </a>
  </div>
  <div
    className="col-4"
  >
    <a
      href="https://www.instagram.com/tarkanmansuroglu/"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-instagram fa-3x contactInstagram"
      />
    </a>
  </div>
  <div
    className="col-4"
  >
    <a
      href="https://twitter.com/T_Mansuroglu"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-twitter fa-3x contactTwitter"
      />
    </a>
  </div>
  <div
    className="col-4"
  >
    <i
      className="fas fa-mobile-alt fa-3x"
      onClick={[Function]}
    />
  </div>
  <div
    className="col-4"
  >
    <i
      className="far fa-envelope fa-3x"
      onClick={[Function]}
    />
  </div>
</div>
`;
